feat(howto): add howto use a custom VHAL with Anbox Cloud

ajanon opened the pull request:
This covers how to use the ODM partition and an addon to override the Anbox Cloud VHAL implementation with a custom-built one. This doesn't cover how to build such a binary, and that's left to the reader for now.

keirthana commented on howto/android/odm_vhal.md:
```diff
@@ -0,0 +1,128 @@
+implementation with your own implementation placed in the
+[ODM partition](https://source.android.com/docs/core/architecture/partitions/odm-partitions)
+using an [addon](https://anbox-cloud.io/docs/explanation/addons).
+
```
Adding a heading as "##Prerequisites" may make this more prominent.

keirthana commented on howto/android/odm_vhal.md:
```diff
@@ -0,0 +1,128 @@
+property to `odm`. This will be done by writing that system property to the
+`/odm/etc/build.prop` file.
+
+In the following, we will name our addon `custom-vhal`.
```
Did you mean to say "In the following example"?

ajanon replied:
Yeah, that seems better, thanks!

keirthana commented on howto/android/odm_vhal.md:
```diff
@@ -0,0 +1,128 @@
+1. Copying the VHAL binary to `/odm/bin/hw`
+1. Copying an
+[init file](https://android.googlesource.com/platform/system/core/+/master/init/README.md)
+for your VHAL service to `/odm/etc/init`
```
I would recommend to use period at the end of all points to be consistent.

keirthana commented on howto/android/odm_vhal.md:
````diff
@@ -0,0 +1,128 @@
+└── vhal.rc
+```
+
+`vhal` is the VHAL implementation built in the previous section. If your VHAL
````
"built in the previous section" is a bit confusing for me as we have not provided instructions to build it. IMO, it is a prerequisite to have a cusom implementation ready to do this procedure. Let me know if my understanding is wrong.

```suggestion
`vhal` is your custom VHAL implementation.
```

ajanon replied:
Oh you're right, that's a leftover from a previous version where I assumed I would have instructions for building the binary... Thanks for catching this!

keirthana commented on howto/android/odm_vhal.md:
```diff
@@ -0,0 +1,128 @@
+`vhal` is the VHAL implementation built in the previous section. If your VHAL
+implementation requires additional libraries, they must also be bundled with the
+addon and copied in the ODM partition.
+
```
The following can also be part of the prerequisites section, the idea is to have everything the user needs to do *before* the actual steps of a how-to guide in the prerequisites/preparation section.

" If your VHAL implementation requires additional libraries, they must also be bundled with the
addon and copied in the ODM partition."

ajanon replied:
Makes sense. I will add a small reminder here to ensure people remember.

ajanon replied:
Please tell me if it's better now. I can also remove the small reminder if you think it's redundant.

keirthana replied:
Looks good now.

keirthana commented on howto/android/odm_vhal.md:
````diff
@@ -0,0 +1,128 @@
+  - custom-vhal
+```
+
+## Additional information
````
A very minor nit, pardon me, but I am trying to keep things as consistent as possible:

```suggestion
## Related information
```

keirthana reviewed (changes requested):
First of all, this is excellent documentation in terms of clarity, flow of thought, explanation for all necessary terms and steps. Really great job! 

Some general observations:
- You may have already noticed from the GH actions but some acronyms need to be added to the .wordlist.txt and also links needs to be discourse links.
- I can do this while publishing as well but noting it here just for the sake of completion: We need to add page entries to the navigation table in `index.md` .

ajanon commented:
> First of all, this is excellent documentation in terms of clarity, flow of thought, explanation for all necessary terms and steps. Really great job!

Thanks! :)

> Some general observations:
> 
> * You may have already noticed from the GH actions but some acronyms need to be added to the .wordlist.txt and also links needs to be discourse links.
> 
> * I can do this while publishing as well but noting it here just for the sake of completion: We need to add page entries to the navigation table in `index.md` .

I've fixed this in the latest revision, and I think I addressed all your comments. Thanks for the review!

adglkh commented on howto/android/odm_vhal.md:
````diff
@@ -0,0 +1,133 @@
+
+`vhal.rc` is an [init.rc file](https://android.googlesource.com/platform/system/core/+/main/init/README.md) for your VHAL service, such as:
+
+```
````
Maybe it's worth adding a condition here to start the custom-vhal only when `ro.anbox.automotive.vhal` is set to odm. Otherwise, the default vhal implementation we provided may launch alongside custom-vhal.

ajanon replied:
Hum, that shouldn't happen as `ro.anbox.automotive.vhal` would always be set to `odm` by `/odm/etc/build.prop`

I agree that it's a bit safer, though, I will add it.

morphis commented on howto/android/odm_vhal.md:
```diff
@@ -0,0 +1,137 @@
+*since 1.22.0*
+
+[note type="information" status="Note"]Replacing the Anbox VHAL is only supported on AAOS images.
```
At best we link to the image overview here to allow people identify these easily.

morphis commented on howto/android/odm_vhal.md:
````diff
@@ -0,0 +1,137 @@
+```
+
+Please note that due to Snap strict confinement, the addon directory
+(`custom-vhal` here) must be located in the home directory of the user.
````
```suggestion
(`custom-vhal` here) must be located in the home directory of the user executing the `amc` command.
```

morphis reviewed (changes requested):
Looks really good, just a few small things.
